File: proga/1_semestr/labs/lab3/src/main/java/dev/lysmux/lab3/entity/ability/GlowsInDark.java
package dev.lysmux.lab3.entity.ability;

import java.util.Objects;

public class GlowsInDark extends Ability {
    @Override
    public String getDescription() {
        return toString();
    }

    @Override
    public String toString() {
        return "светиться в темноте";
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (!(o instanceof GlowsInDark glowsInDark)) return false;
        if (!super.equals(o)) return false;
        return Objects.equals(glowsInDark.toString(), glowsInDark.toString());
    }

    @Override
    public int hashCode() {
        return Objects.hash(super.hashCode(), toString());
    }
}

File: proga/1_semestr/labs/lab3/src/main/kotlin/dev/lysmux/lab3/entity/ability/GlowsInDark.kt
package dev.lysmux.lab3.entity.ability

class GlowsInDark : Ability() {
    override fun getDescription() = toString()

    override fun toString() = "светиться в темноте"

    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        return other is GlowsInDark
    }

    override fun hashCode() = javaClass.hashCode()
}